src/core/qgsvectorlayer.h

```diff
@@ -2603,6 +2602,9 @@ class CORE_EXPORT QgsVectorLayer : public QgsMapLayer, public QgsExpressionConte
     // Feature counts for each renderer legend key
     QHash<QString, long> mSymbolFeatureCountMap;
 
+    // feature id for each renderer legend keyh
+    QHash<QString, QgsFeatureIds> mSymbolIdMap;
```

Comment:
What's this still used for?

Reply:
Storing the fids for each symbol. Needed to provide to the filters during expression evaluation.

Reply:
I only see this member being updated (ids added, map cleared) but never actually used. Can you point me to where that happens?

Reply:
See line 3925 of the c++ file. I forgot to change it after the fids filter merge.